guides/create_an_app/4-query.md

```diff
@@ -1,5 +1,91 @@
 # 4. Query
 
-Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin nibh augue, suscipit a, scelerisque sed, lacinia in, mi. Cras vel lorem. Etiam pellentesque aliquet tellus. Phasellus pharetra nulla ac diam. Quisque semper justo at risus. Donec venenatis, turpis vel hendrerit interdum, dui ligula ultricies purus, sed posuere libero dui id orci. Nam congue, pede vitae dapibus aliquet, elit magna vulputate arcu, vel tempus metus leo non est. Etiam sit amet lectus quis est congue mollis. Phasellus congue lacus eget neque. Phasellus ornare, ante vitae consectetuer consequat, purus sapien ultricies dolor, et mollis pede metus eget nisi. Praesent sodales velit quis augue. Cras suscipit, urna at aliquam rhoncus, urna quam viverra nisi, in interdum massa nibh nec erat.
+## Retrieving from database
+
+Every time, when refreshing your page, you lose the message history. This is not an ideal situation when you want to see what has been written. We have all these stored messages in our database and we want to retrieve them when someone logs in. Therefore, we will need to retrieve these messages and display them in the chat dialogue.
+
+## Perform a callback `after_join`.
+Go to the join function, and add the line just below `if authorized?(payload) do`
+
+We are going to send a message contaning our `self` *PID(Process Identifier)* as the first parameter. The second parameter is a message to run a function called `after_join`. This function is identified by an atom type.
+Add this line to function inside `lib/workshops_app_web/channels/general_channel.ex`
+
+```elixir
+  def join("random:lobby", payload, socket) do
+    if authorized?(payload) do
+      send(self(), :after_join) # <-- Add this line
+      {:ok, socket}
+    else
+      {:error, %{reason: "unauthorized"}}
+    end
+  end
+```
+
+### What does send do?
+`send` sends a message `:after_join`, with it's PID generated by the function `self()`. We want to be able to run that after join task asynchronously. Lets look at how we should receive the `:after_join` message.
+
+## Receiving a message in a callback
+We now know that a message identified by the atom `:after_join` is being sent. Phoenix provides us with a function called `handle_info` to be able to receive that message and do something with it. Insert the following code after the `join` function.
+
+```elixir
+  def handle_info(:after_join, socket) do
+    messages =
+      WorkshopsApp.Conversations.list_messages()
+      |> Enum.map(fn(%{message: message, name: name}) -> %{message: message, name: name} end)
+    push socket, "messages_history", %{messages: messages}
+    {:noreply, socket}
+  end
+```
+
+**Coach:** Please explain concept and above example of pattern matching.
+
+**Coach:** Please explain `&` operator for function shorthand.
```

Comment:
I would remove this. It's really not important in the beginning. In addition, IMO two coach tips, one by one is too much.

Reply:
you underestimate our coaches :P

Reply:
but I agree, removed and merging